internal/wasm/table.go

```diff
@@ -308,3 +309,29 @@ func (m *Module) verifyImportGlobalI32(sectionID SectionID, sectionIdx Index, id
 	}
 	return fmt.Errorf("%s[%d] (global.get %d): out of range of imported globals", SectionIDName(sectionID), sectionIdx, idx)
 }
+
+// Grow appends the `initialRef` by `delta` times into the .References slice.
+// Returns -1 if the operation is not valid, otherwise the old length of the table.
+//
+// https://www.w3.org/TR/2022/WD-wasm-core-2-20220419/exec/instructions.html#xref-syntax-instructions-syntax-instr-table-mathsf-table-grow-x
+func (t *TableInstance) Grow(_ context.Context, delta uint32, initialRef Reference) (currentLen uint32) {
+	currentLen = uint32(len(t.References))
```

Comment:
since tables can be shared (imported by N modules) maybe re-use same lock strategy as mem on grow overlap?